src/frontend/src/player.ts

```diff
@@ -27,252 +63,368 @@ export class Player {
-              newStreamData.audio.currentStream = this.currentStreams.audio.currentStream
-              this.currentStreams = newStreamData
-            }
-          } else if (this.currentStreams.audio && this.currentStreams.audio.tracks) {
-            this.currentStreams = newStreamData
-          }
-          this.updateChannelData(this.currentStreams)
+  async stop(): Promise<void> {
+    await this.stopping
+    this.stopping = new Promise(async (resolve) => {
+      try {
+        clearTimeout(this.replayOnErrorTimeout)
+        if (this.streams) {
+          await this.terminateStreams(this.streams)
+          this.streams = null
         }
-      })
-    } else {
-      ffprobe(`${url}?timeout=${ffprobeTimeout}`, { path: ffprobePath }, (err: Error, metadata: any) => {
-        if (metadata) {
-          this.currentStreams = this.processStreams(metadata.streams, url)
-          this.updateChannelData(this.currentStreams)
-          this.playUrl(this.currentStreams.url, this.currentStreams, evt)
-
-        } else {
-          console.log('ffprobe failure')
+        if (this.udpStreamer) {
+          await this.udpStreamer.close()
         }
-      })
+      } catch (e) {
+        console.error('An error occurred while stopping:', e)
+      } finally {
+        resolve()
+      }
+    })
+    return this.stopping
+  }
+
+  async terminateStreams(streams: IPlayerStreams): Promise<void> {
+    if (streams.input instanceof Readable) {
+      await this.destroyStream(streams.input)
     }
+    await this.killFfmpeg(streams.ffmpeg)
+    await this.destroyStream(streams.pipeline)
+  }
+
+  // We need to wait a bit for the process to terminate
+  // this prevents ffmpeg to catch output stream "close" event and throw an error
+  killFfmpeg(process: FfmpegCommand): Promise<void> {
+    return new Promise((resolve) => {
+      process.on('error', resolve)
+      process.kill('SIGKILL')
+    })
   }
 
-  handleConversionError (evt: any) {
-        // if conversion error, keep trying this.playUrl will kill previous process
-    this.playUrl(this.currentStreams.url, this.currentStreams, evt)
+  destroyStream(stream: Readable | Writable): Promise<void> {
+    return new Promise((resolve) => {
+      stream.on('close', resolve)
+      stream.destroy()
+    })
   }
 
-  handleErroneousStreamError(evt: any) {
-    this.currentStreams.audio.currentStream = (this.currentStreams.audio.tracks.length > 0) ? this.currentStreams.audio.tracks[0].pid :-1
-    this.currentStreams.subtitles.currentStream = -1
-    this.playUrl(this.currentStreams.url, this.currentStreams, evt)
+  async retrieveMetadata(url: string): Promise<Ffmpeg.FfprobeData> {
+    let input: string | Readable
 
+    if (this.decryption.use) {
+      input = await this.getDecryptionPipeline(url)
+    } else {
+      input = url
+    }
+
+    try {
+      const metadata = await ffprobe(input, { timeout: 30 })
```

Comment:
```suggestion
      return await ffprobe(input, { timeout: 30 })
```

Reply:
I'm trying to use a try / catch block here, if I return a Promise that fails it won't be properly caught with this.